src/parser/syntax.rs

```diff
@@ -431,10 +477,34 @@ pub mod tests {
         P: Parser<Span<'a>, R, ParseError<Span<'a>, Scalar>>,
         R: std::fmt::Display + std::fmt::Debug + Clone + Eq,
     {
+        // NOTE: Please do not try to refactor this match to make it simpler or
+        // or remove the commented printlns, which are to aid in debugging the parser
         match (expected, p.parse(Span::<'a>::new(i))) {
-            (Some(expected), Ok((_, x))) if x == expected => true,
-            (Some(_) | None, Ok(..)) | (Some(..), Err(_)) => false,
-            (None, Err(_e)) => true,
+            (Some(expected), Ok((_, x))) if x == expected => {
+                println!("Expected: {expected}");
+                println!("Detected: {x}");
```

Comment:
Are these the 'commented printlns' mentioend above? I think they need to be commented out if so.